diagnosis/k8sCollectLogsCi.sh

```diff
@@ -200,32 +188,39 @@ then
     scp ${SCP_FLAGS} hosts.sh ${USER}@${MASTER_IP}:/home/${USER}/hosts.sh
     ssh ${SSH_FLAGS} ${USER}@${MASTER_IP} "sudo chmod 744 hosts.sh; ./hosts.sh"
     scp ${SCP_FLAGS} ${USER}@${MASTER_IP}:/home/${USER}/cluster-snapshot.zip ${LOGFILEFOLDER}/cluster-snapshot.zip
-    ssh ${SSH_FLAGS} ${USER}@${MASTER_IP} "kubectl get nodes -o json|jq -r '.items[].metadata.name' | grep -v  k8s-master-* > cluster_nodes.txt"
-    scp ${SCP_FLAGS} ${USER}@${MASTER_IP}:/home/${USER}/cluster_nodes.txt ${LOGFILEFOLDER}/cluster-nodes.txt
-    ssh ${SSH_FLAGS} ${USER}@${MASTER_IP} "sudo rm -f cluster-snapshot.zip hosts.sh cluster_nodes.txt"
+    scp ${SCP_FLAGS} ${USER}@${MASTER_IP}:/home/${USER}/*_nodes.txt ${LOGFILEFOLDER}/.
+    ssh ${SSH_FLAGS} ${USER}@${MASTER_IP} "sudo rm -f cluster-snapshot.zip hosts.sh *_nodes.txt"
     
-    if [ ! -f ${LOGFILEFOLDER}/cluster-nodes.txt ]
+    if [ ! -f ${LOGFILEFOLDER}/linux_nodes.txt ]
     then
-        echo "Agent nodes not present"
+        echo "Linux nodes not present"
     else
         PROXY_CMD="ssh -i ${IDENTITYFILE} ${KNOWN_HOSTS_OPTIONS} ${USER}@${MASTER_IP} -W %h:%p"
         
-        INPUT_FILE=${LOGFILEFOLDER}/cluster-nodes.txt
-        CLUSTER_NODES=$(<$INPUT_FILE)
+        INPUT_FILE=${LOGFILEFOLDER}/linux_nodes.txt
+        LINUX_NODES=$(<$INPUT_FILE)
         
-        for host in ${CLUSTER_NODES}
+        for host in ${LINUX_NODES}
         do
             processHost ${host}
         done
     fi
-fi
 
-echo "Removing master files copied"
-if [ -f ${LOGFILEFOLDER}/aksMasterIdentityFile ]
-then
-    echo "Removing master files copied"
-    rm ${LOGFILEFOLDER}/aksMasterIdentityFile
+    if [ ! -f ${LOGFILEFOLDER}/windows_nodes.txt ]
+    then
+        echo "Windows nodes not present"
+    else
+        PROXY_CMD="ssh -i ${IDENTITYFILE} ${KNOWN_HOSTS_OPTIONS} ${USER}@${MASTER_IP} -W %h:%p"
+        
+        INPUT_FILE=${LOGFILEFOLDER}/windows_nodes.txt
+        WINDOWS_NODES=$(<$INPUT_FILE)
+        
+        for host in ${WINDOWS_NODES}
+        do
+            processHost ${host}
+        done
+    fi
 fi
 
-echo "[$(date +%Y%m%d%H%M%S)][INFO] Logs can be found here: $LOGFILEFOLDER"
+echo "[$(date +%Y%m%d%H%M%S)][INFO] Done with k8s log collection"
```

Comment:
it may be useful to print the output directory

Reply:
Updated